Gulp/Deploy Broken on Windows

By adding `electron-installer-debian` to devDependencies in de56f180d81bb0deb2ff82f6807d57dadfe8ea42, installation is now broken on Windows. npm gives the following error message:
```
npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for electron-installer-debian@3.1.0: wanted {"os":"darwin,linux","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm ERR! notsup Valid OS:    darwin,linux
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   win32
npm ERR! notsup Actual Arch: x64
```

To fix this, `electron-installer-debian` should be moved to `optionalDependencies`, and `require('electron-installer-debian'` should be replaced with `try require('electron-installer-debian')`. This is exactly how `electron-installer-flatpak` is handled presently.

Let me know, and I can make a PR for this change.

--- a/gulpfile.coffee
+++ b/gulpfile.coffee
@@ -13,11 +13,11 @@ changed    = require 'gulp-changed'
 rename     = require 'gulp-rename'
 packager   = require 'electron-packager'
 flatpak    = try require 'electron-installer-flatpak'
+debian     = try require 'electron-installer-debian'
 filter     = require 'gulp-filter'
 Q          = require 'q'
 Stream     = require 'stream'
 spawn      = require('child_process').spawn
-debian     = require('electron-installer-debian')
 
 #
 #

--- a/gulpfile.coffee
+++ b/gulpfile.coffee
@@ -236,6 +236,7 @@ names = {linux: [], win32: [], darwin: []}
 #
 
 zipIt = (folder, filePrefix, done) ->
+    # use zip
     ext = 'zip'
     zipName = path.join outdeploy, "#{filePrefix}.#{ext}"
     folder = path.basename folder
@@ -248,11 +249,25 @@ zipIt = (folder, filePrefix, done) ->
     compressIt('zip', args, opts, zipName, done)
 
 tarIt = (folder, filePrefix, done) ->
+    # use GNU tar to make gzipped tar archive
     ext = 'tar.gz'
     zipName = path.join outdeploy, "#{filePrefix}.#{ext}"
     folder = path.basename folder
     #
     args = ['-czf', zipName, folder]
+    opts = {
+        cwd: outdeploy
+        stdio: [0, 1, 'pipe']
+    }
+    compressIt('tar', args, opts, zipName, done)
+
+zipItWin = (folder, filePrefix, done) ->
+    # use built-in tar.exe to make zip archive
+    ext = 'zip'
+    zipName = path.join outdeploy, "#{filePrefix}.#{ext}"
+    folder = path.basename folder
+    #
+    args = ['-cf', zipName, folder]
     opts = {
         cwd: outdeploy
         stdio: [0, 1, 'pipe']
@@ -316,6 +331,10 @@ deploy = (platform, arch, cb) ->
 
                 if platform == 'linux'
                     tarIt zippath, fileprefix, ->
+                      cb()
+                      deferred.resolve()
+                else if platform == 'win32'
+                    zipItWin zippath, fileprefix, ->
                       cb()
                       deferred.resolve()
                 else

--- a/package.json
+++ b/package.json
@@ -75,14 +75,14 @@
     "uber-url-regex": "^5.0.9"
   },
   "optionalDependencies": {
-    "electron-installer-flatpak": "^0.6.0"
+    "electron-installer-flatpak": "^0.6.0",
+    "electron-installer-debian": "^3.1.0"
   },
   "devDependencies": {
     "chai": "^4.2.0",
     "coffeescript": "^2.5.1",
     "devtron": "^1.4.0",
     "electron": "^9.1.0",
-    "electron-installer-debian": "^3.1.0",
     "electron-packager": "^15.0.0",
     "gulp": "^4.0.2",
     "gulp-changed": "^4.0.2",

